src/main/java/uk/gov/hmcts/ccd/domain/service/casedeletion/CaseDataExtractor.java

```diff
@@ -0,0 +1,187 @@
+
+@Named
+@Slf4j
+public class CaseDataExtractor {
+    private static final String EMPTY_STRING = "";
+    private static final String FIELD_SEPARATOR = ".";
+
+    public List<String> extractFieldTypePaths(final Map<String, JsonNode> data,
+                                              final List<CaseFieldDefinition> caseFieldDefinitions,
+                                              final String fieldType) {
+        return extractFieldTypePaths(
+            data,
+            caseFieldDefinitions,
+            CaseDataExtractor.EMPTY_STRING,
+            new ArrayList<>(),
+            fieldType);
+    }
+
+    public List<String> extractFieldTypePaths(final Map<String, JsonNode> data,
+                                              final List<CaseFieldDefinition> caseFieldDefinitions,
+                                              final String fieldIdPrefix,
+                                              List<String> paths,
+                                              final String fieldType) {
+        return (data == null)
+            ? Collections.emptyList()
+            : data.entrySet().stream()
+                .map(caseDataPair -> caseFieldDefinitions.stream()
+                    .filter(caseField -> caseField.getId().equalsIgnoreCase(caseDataPair.getKey()))
+                    .findAny()
+                    .map(caseField -> extractField(
+                        caseDataPair.getKey(),
+                        caseDataPair.getValue(),
+                        caseField,
+                        fieldIdPrefix,
+                        paths,fieldType
+                    ))
+                    .orElseGet(Collections::emptyList))
+            .flatMap(List::stream)
+            .collect(Collectors.toList());
+    }
+
+    private List<String> extractField(final String dataFieldId,
+                                                final JsonNode dataValue,
+                                                final CaseFieldDefinition caseFieldDefinition,
+                                                final String fieldIdPrefix,
+                                                List<String> paths,
+                                                String fieldType) {
+        final String caseFieldType = caseFieldDefinition.getFieldTypeDefinition().getType();
+
+        if (!BaseType.contains(caseFieldType)) {
+            log.debug("Ignoring Unknown Type: " + caseFieldType);
+            return Collections.emptyList();
+        }
+
+        final BaseType baseFieldType = BaseType.get(caseFieldType);
+
+        if (BaseType.get(COMPLEX) == baseFieldType) {
+            return extractFieldTypePaths(
+                JacksonUtils.convertValue(dataValue),
+                caseFieldDefinition.getFieldTypeDefinition().getComplexFields(),
+                fieldIdPrefix + dataFieldId + FIELD_SEPARATOR, paths, fieldType
+            );
+        } else if (BaseType.get(COLLECTION) == baseFieldType) {
```

Comment:
Can this `else if` go in an `#extractCollectionField(...)` just like `#extractSimpleField(...)`

Reply:
Done [c129a9a](https://github.com/hmcts/ccd-data-store-api/pull/1872/commits/c129a9a6a6fdff4226e65e0b027691d3cd49352f)